lib/register.php

```diff
@@ -272,9 +272,21 @@ function gutenberg_can_edit_post( $post_id ) {
  * @param int $post_id Post ID.
  * @return bool Whether the post has blocks.
  */
-function gutenberg_post_has_blocks( $post_id ) {
+function post_has_blocks( $post_id ) {
 	$post = get_post( $post_id );
-	return $post && strpos( $post->post_content, '<!-- wp:' ) !== false;
+	return $post && content_has_blocks( $post->post_content );
+}
+
+/**
+ * Determine whether a content string contains blocks.
+ *
+ * @since 1.2.0
```

Comment:
do you think it would be worth adding a note on the limited ability this has to accurately determine blocks? I would think it obvious from looking at the code but maybe someone wouldn't realize this is only a first-order approximation.

Reply:
Yeah, could be a good idea. The implementation is dubious, but... do you have in mind any scenarios in which it would not be accurate?

Reply:
just the rare case where someone writes something like…

```
This is not a <!-- wp --> block. <!-- wp:comment -->
```

…or when something odd like that accidentally crops up.